rero_ils/modules/items/api/api.py

```diff
@@ -249,16 +239,58 @@ def index(self, record):
                 except IlsRecordError.NotDeleted:
                     pass
 
+    def get_old_holdings_pid(self, pid):
+        """Get the old holding record if exists.
+
+        :param pid: pid of record.
+        """
+        items_search = ItemsSearch(). \
+            filter('term', pid=pid). \
+            source('holding').execute().hits
+
+        old_holdings_pid = None
+        if items_search.total.value:
+            old_holdings_pid = items_search[0].holding.pid
+        return old_holdings_pid
+
+    def index(self, record):
+        """Index an item."""
+        old_holdings_pid = self.get_old_holdings_pid(record.pid)
+        return_value = super().index(record)
+        self.items_after_index_record(record, old_holdings_pid)
```

Comment:
why not using: `after_index_record`?

Reply:
The parameters are not the same. We need `old_holdings_pid` to check if old holding can be deleted.